### ISSUE TITLE
"File->New" should remember last setup

### ISSUE THREAD

**Issue by FelixBrendel** [yishn/Sabaki#124]

**Comment by yishn** [comment]
Can you elaborate what you mean exactly by 'last setup'?

**Comment by FelixBrendel** [comment]
I'd find it very helpful if the window that comes up when you select File->new already has entries in it (the ones that were use for the last game), so you don't have to enter everything again. In the current version you have to enter everything, so you could reduce the amount of writing one has to do to start a new game. I for myself for example like playing quite a few games against the same bot on the same size board (9x9 or 12x12). So with the improvement I'm trying to describe, I wont have to set the bot as my opponent every new game, write my name in the name box (I know that's just optional, but nice to have) and set the board size. 
Implementation-wise I'd see to options: You could save the last entries just in system memory and use it as long as the program is running, or write to a file so even after the program is closed the settings for the last game will stored.

**Comment by yishn** [comment]
I'm wondering what fields should be remembered? Maybe:

- Engine
- Komi
- Handicap
- Board Size

I don't think anyone would want the game name, event name or date to be remembered. Furthermore, should Sabaki override the data if one opens a SGF file? Or only save the data when the user creates new games? 

Another idea is to add a 'Save As Defaults' button in the game info drawer. When the user creates a new game, the default settings will be applied. What do you think?

**Comment by FelixBrendel** [comment]
> I'm wondering what fields should be remembered?

Yes these are the main points, but I'd also add the **player's name** and possibly even **Event** to that. If you are "recording" a game from an event, the event name will stay the same, otherwise it will be empty since empty was stored before.

Also I think, if you open a SGF file, your settings **shouldn't be overwritten**, since when you create a new file you probably want to continue **your own work**/writing down, and not for example a joseki-library or a tournament game.

To save the setup is definitely a great option, but in that case I would rather be able to create multiple setups, not just one ("default settings"), and then be able to chose from my presets in the File->New dialog. 

What do you think about it?

**Comment by yishn** [comment]
I think having multiple presets is out of Sabaki's scope right now. Frankly speaking, I'm not sure if remembering the settings from previous created games is a good idea. Consider the following scenario: 

Users can create multiple games within one file. By clicking `File` > `Manage Games...` > `Add` > `Add New Game` you can add a 'blank game' to the collection. Sabaki won't show the game info drawer this way. If Sabaki chooses to create a game from its remembered settings, the user may not even realize.

**Comment by FelixBrendel** [comment]
In this case I'd just use the stored date only in the File->New menu, when it's visible to the user.
By the way can you speak German? This way conversation was much easier and more informative I think ^^

**Comment by FelixBrendel** [comment]
Now that I thought about it a bit longer I think the New Game dialog should also come up when you select File->Manage Games...->-Add->Add New. That way you can give games names right away, which is always important to stay organized. Otherwise you have to manually open menu->Info to enter a name. And if people don't want to enter a name or other infos they can just click OK because then the last settings will be used. Which most likely will have the same settings.

**Comment by TheIronMarx** [comment]
The presets would be a nice feature, but are perhaps out of scope for the project currently. I do, however, believe that if I'm playing multiple games with somebody locally, I don't want to have to set date, event, names, ranks, and most critically board size and handicap. Every time I hit File > New. If I close the program, then by all means don't persist that information. But if I'm in one session, I don't find it unreasonable to persist at least board size and handicap.

**Comment by yishn** [comment]
I'm still not sure about this 'Remember game info in a session' thing. It's just a bit of an unexpected behavior. One would think that the game info dialog has only effect on the *current* game, but instead, it has some *temporary* global side effects.

Also, the game info dialog should behave the same, no matter how it's opened. However, @FelixBrendel proposes, that opening the game info dialog (and clicking 'OK') on a game that the user didn't create for himself should *not* persist information. (This wouldn't be a problem if we just remember board size and handicap, because these fields are disabled in a non-empty game)

I understand it may be tiresome to type in the names, ranks, event in both of your situations. But there are definitely other cases where having the previous names, ranks, event filled in is not beneficial and tedious to *remove*.

**TL;DR** I'm okay with persisting board size and handicap *forever*. But persisting anything else, or anything only within a session, is IMO unreasonable.

**Comment by yishn** [comment]
I'm still not sure about this 'Remember game info in a session' thing. It's just a bit of an unexpected behavior. One would think that the game info dialog has only effect on the *current* game, but instead, it would have some *temporary* global side effects.

The game info dialog should behave the same, no matter how it's opened. However, @FelixBrendel proposes, that opening the game info dialog (and clicking 'OK') on a game that the user didn't create for himself should *not* persist information. (This wouldn't be a problem if we just remember board size and handicap, because the user can't change them in a non-empty game)

I understand it may be tiresome to type in the names, ranks, event in both of your situations. But there are definitely other cases where having the previous names, ranks, event filled in is not beneficial and tedious to *remove*.

**TL;DR** I'm okay with persisting board size and handicap *forever*. But persisting anything else, or anything only within a session, is IMO unreasonable.
